The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15065916493/job/42350672190) [(plain)](https://github.com/rust-lang/rust/commit/4c2d3bc8bd663d619ae8502bc62afb49efb0ddfa/checks/42350672190/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tests/ui/ref_option/ref_option.all.fixed ... ok
tests/ui/non_std_lazy_static/non_std_lazy_static_fixable.fixed ... ok

FAILED TEST: tests/ui/map_flatten.rs
command: CLIPPY_CONF_DIR="tests" RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Ainternal_features" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Dwarnings" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps" "--extern=clippy_config=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_config-9788b7065cd214bc.rlib" "--extern=clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_lints-d7d57cd954d7e14c.rlib" "--extern=clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_utils-915e1e1e8bd73bb2.rlib" "--extern=futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libfutures-5a7156af45221746.rlib" "--extern=if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libif_chain-d11c9c2ffe481e38.rlib" "--extern=itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libitertools-3e76c5e4cf3f0bdf.rlib" "--extern=parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libparking_lot-e095505c7235a9a3.rlib" "--extern=quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libquote-c8178baeef41ae19.rlib" "--extern=regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libregex-cb003e5d2ad3f3b6.rlib" "--extern=serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libserde-a43a28edafb16747.rlib" "--extern=serde_derive=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/libserde_derive-9780bbb027e4cda0.so" "--extern=syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libsyn-e6d09ab535c31563.rlib" "--extern=tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libtokio-85c5ed73f0cfcc77.rlib" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/release/deps" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui_test/0/tests/ui" "tests/ui/map_flatten.rs" "--edition" "2024"

error: actual output differed from expected
Execute `./x test src/tools/clippy --bless` to update `tests/ui/map_flatten.stderr` to the actual output
--- tests/ui/map_flatten.stderr
+++ <stderr output>
---
+   |
+   = note: `-D stable-features` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(stable_features)]`
+
 error: called `map(..).flatten()` on `Option`
   --> tests/ui/map_flatten.rs:8:10
... 110 lines skipped ...
    | |__________________^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| x.iter())`
 
-error: aborting due to 5 previous errors
+error: aborting due to 6 previous errors
 

---
   |
   = note: `-D stable-features` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(stable_features)]`

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten.rs:8:10
   |
LL |           .map(|x| {
   |  __________^
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
   = note: `-D clippy::map-flatten` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::map_flatten)]`
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|x| {
LL +
LL + 
LL + 
LL +             if x <= 5 {
LL +                 Some(x)
LL +             } else {
LL +                 None
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten.rs:21:10
   |
LL |           .map(|x| {
   |  __________^
LL | |
LL | |
LL | |             if x == 1 {
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|x| {
LL +
LL + 
LL +             if x == 1 {
LL +                 Ok(x)
LL +             } else {
LL +                 Err(0)
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten.rs:35:10
   |
LL |           .map(|res| {
   |  __________^
LL | |
LL | |
LL | |             if res > 0 {
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|res| {
LL +
LL + 
LL +             if res > 0 {
LL +                 do_something();
LL +                 Ok(res)
LL +             } else {
LL +                 Err(0)
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten.rs:49:10
   |
LL |           .map(|some_value| {
   |  __________^
LL | |
LL | |
LL | |             if some_value > 3 {
...  |
LL | |         })
LL | |         .flatten()
   | |__________________^
   |
help: try replacing `map` with `filter_map` and remove the `.flatten()`
   |
LL ~         .filter_map(|some_value| {
LL +
LL + 
LL +             if some_value > 3 {
LL +                 Some(some_value)
LL +             } else {
LL +                 None
LL +             }
LL +         })
   |

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten.rs:68:10
   |
LL |           .map(|x| x.iter())
   |  __________^
...  |
LL | |         .flatten();
   | |__________________^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| x.iter())`

error: aborting due to 6 previous errors



error: there were 1 unmatched diagnostics
##[error] --> tests/ui/map_flatten.rs:2:12
  |
2 | #![feature(result_flattening)]
  |            ^^^^^^^^^^^^^^^^^ Error[stable_features]: the feature `result_flattening` has been stable since 1.89.0-nightly and no longer requires an attribute to enable
  |

full stderr:
error: the feature `result_flattening` has been stable since 1.89.0-nightly and no longer requires an attribute to enable
##[error]  --> tests/ui/map_flatten.rs:2:12
   |
LL | #![feature(result_flattening)]
   |            ^^^^^^^^^^^^^^^^^
   |
   = note: `-D stable-features` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(stable_features)]`

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten.rs:8:10
   |
LL |           .map(|x| {
   |  __________^
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
   = note: `-D clippy::map-flatten` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::map_flatten)]`
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|x| {
LL +
LL + 
LL + 
LL +             if x <= 5 {
LL +                 Some(x)
LL +             } else {
LL +                 None
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten.rs:21:10
   |
LL |           .map(|x| {
   |  __________^
LL | |
LL | |
LL | |             if x == 1 {
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|x| {
LL +
LL + 
LL +             if x == 1 {
LL +                 Ok(x)
LL +             } else {
LL +                 Err(0)
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten.rs:35:10
   |
LL |           .map(|res| {
   |  __________^
LL | |
LL | |
LL | |             if res > 0 {
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|res| {
LL +
LL + 
LL +             if res > 0 {
LL +                 do_something();
LL +                 Ok(res)
LL +             } else {
LL +                 Err(0)
LL +             }
LL ~         });
   |

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten.rs:49:10
   |
LL |           .map(|some_value| {
   |  __________^
LL | |
LL | |
LL | |             if some_value > 3 {
...  |
LL | |         })
LL | |         .flatten()
   | |__________________^
   |
help: try replacing `map` with `filter_map` and remove the `.flatten()`
   |
LL ~         .filter_map(|some_value| {
LL +
LL + 
LL +             if some_value > 3 {
LL +                 Some(some_value)
LL +             } else {
LL +                 None
LL +             }
LL +         })
   |

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten.rs:68:10
   |
LL |           .map(|x| x.iter())
   |  __________^
...  |
LL | |         .flatten();
   | |__________________^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| x.iter())`

error: aborting due to 6 previous errors


full stdout:



FAILED TEST: tests/ui/map_flatten_fixable.rs
command: CLIPPY_CONF_DIR="tests" RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Ainternal_features" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Dwarnings" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps" "--extern=clippy_config=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_config-9788b7065cd214bc.rlib" "--extern=clippy_lints=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_lints-d7d57cd954d7e14c.rlib" "--extern=clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_utils-915e1e1e8bd73bb2.rlib" "--extern=futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libfutures-5a7156af45221746.rlib" "--extern=if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libif_chain-d11c9c2ffe481e38.rlib" "--extern=itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libitertools-3e76c5e4cf3f0bdf.rlib" "--extern=parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libparking_lot-e095505c7235a9a3.rlib" "--extern=quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libquote-c8178baeef41ae19.rlib" "--extern=regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libregex-cb003e5d2ad3f3b6.rlib" "--extern=serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libserde-a43a28edafb16747.rlib" "--extern=serde_derive=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/libserde_derive-9780bbb027e4cda0.so" "--extern=syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libsyn-e6d09ab535c31563.rlib" "--extern=tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/libtokio-85c5ed73f0cfcc77.rlib" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/release/deps" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui_test/0/tests/ui" "tests/ui/map_flatten_fixable.rs" "--edition" "2024"

error: actual output differed from expected
Execute `./x test src/tools/clippy --bless` to update `tests/ui/map_flatten_fixable.stderr` to the actual output
--- tests/ui/map_flatten_fixable.stderr
+++ <stderr output>
---
+   |
+   = note: `-D stable-features` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(stable_features)]`
+
 error: called `map(..).flatten()` on `Iterator`
   --> tests/ui/map_flatten_fixable.rs:17:47
... 93 lines skipped ...
    |
 
-error: aborting due to 9 previous errors
---
   |
   = note: `-D stable-features` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(stable_features)]`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:17:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id)`
   |
   = note: `-D clippy::map-flatten` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::map_flatten)]`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:19:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_ref).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_ref)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:21:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_closure).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_closure)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:23:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| x.checked_add(1)).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(|x| x.checked_add(1))`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:27:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| 0..x).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| 0..x)`

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten_fixable.rs:31:40
   |
LL |     let _: Option<_> = (Some(Some(1))).map(|x| x).flatten();
   |                                        ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten_fixable.rs:35:42
   |
LL |     let _: Result<_, &str> = (Ok(Ok(1))).map(|x| x).flatten();
   |                                          ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:45:10
   |
LL |           .map(|n| match n {
   |  __________^
LL | |
LL | |             1 => [n
LL | |                 .saturating_add(1)
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `flat_map` and remove the `.flatten()`
   |
LL ~         .flat_map(|n| match n {
LL +
LL +             1 => [n
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)],
LL +             n => [n],
LL ~         });
   |

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten_fixable.rs:66:10
   |
LL |           .map(|_| {
   |  __________^
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|_| {
LL +
LL + // we need some newlines
LL + // so that the span is big enough
LL + // for a split output of the diagnostic
LL +             Some("")
---
error: there were 1 unmatched diagnostics
##[error] --> tests/ui/map_flatten_fixable.rs:1:12
  |
1 | #![feature(result_flattening)]
  |            ^^^^^^^^^^^^^^^^^ Error[stable_features]: the feature `result_flattening` has been stable since 1.89.0-nightly and no longer requires an attribute to enable
  |

full stderr:
error: the feature `result_flattening` has been stable since 1.89.0-nightly and no longer requires an attribute to enable
##[error]  --> tests/ui/map_flatten_fixable.rs:1:12
   |
LL | #![feature(result_flattening)]
   |            ^^^^^^^^^^^^^^^^^
   |
   = note: `-D stable-features` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(stable_features)]`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:17:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id)`
   |
   = note: `-D clippy::map-flatten` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::map_flatten)]`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:19:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_ref).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_ref)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:21:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_closure).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_closure)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:23:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| x.checked_add(1)).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(|x| x.checked_add(1))`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:27:47
   |
LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| 0..x).flatten().collect();
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| 0..x)`

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten_fixable.rs:31:40
   |
LL |     let _: Option<_> = (Some(Some(1))).map(|x| x).flatten();
   |                                        ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`

error: called `map(..).flatten()` on `Result`
##[error]  --> tests/ui/map_flatten_fixable.rs:35:42
   |
LL |     let _: Result<_, &str> = (Ok(Ok(1))).map(|x| x).flatten();
   |                                          ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`

error: called `map(..).flatten()` on `Iterator`
##[error]  --> tests/ui/map_flatten_fixable.rs:45:10
   |
LL |           .map(|n| match n {
   |  __________^
LL | |
LL | |             1 => [n
LL | |                 .saturating_add(1)
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `flat_map` and remove the `.flatten()`
   |
LL ~         .flat_map(|n| match n {
LL +
LL +             1 => [n
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)
LL +                 .saturating_add(1)],
LL +             n => [n],
LL ~         });
   |

error: called `map(..).flatten()` on `Option`
##[error]  --> tests/ui/map_flatten_fixable.rs:66:10
   |
LL |           .map(|_| {
   |  __________^
...  |
LL | |         })
LL | |         .flatten();
   | |__________________^
   |
help: try replacing `map` with `and_then` and remove the `.flatten()`
   |
LL ~         .and_then(|_| {
LL +
LL + // we need some newlines
LL + // so that the span is big enough
LL + // for a split output of the diagnostic
LL +             Some("")
```

</details>

--- a/src/tools/clippy/tests/ui/map_flatten.rs
+++ b/src/tools/clippy/tests/ui/map_flatten.rs
@@ -1,5 +1,5 @@
 #![warn(clippy::map_flatten)]
-#![feature(result_flattening)]
+
 //@no-rustfix
 // issue #8506, multi-line
 #[rustfmt::skip]

--- a/src/tools/clippy/tests/ui/map_flatten_fixable.fixed
+++ b/src/tools/clippy/tests/ui/map_flatten_fixable.fixed
@@ -1,4 +1,3 @@
-#![feature(result_flattening)]
 #![allow(
     clippy::let_underscore_untyped,
     clippy::missing_docs_in_private_items,

--- a/src/tools/clippy/tests/ui/map_flatten_fixable.rs
+++ b/src/tools/clippy/tests/ui/map_flatten_fixable.rs
@@ -1,4 +1,3 @@
-#![feature(result_flattening)]
 #![allow(
     clippy::let_underscore_untyped,
     clippy::missing_docs_in_private_items,

--- a/src/tools/clippy/tests/ui/map_flatten_fixable.stderr
+++ b/src/tools/clippy/tests/ui/map_flatten_fixable.stderr
@@ -1,5 +1,5 @@
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:17:47
+  --> tests/ui/map_flatten_fixable.rs:16:47
    |
 LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id).flatten().collect();
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id)`
@@ -8,43 +8,43 @@ LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id).flatten().coll
    = help: to override `-D warnings` add `#[allow(clippy::map_flatten)]`
 
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:19:47
+  --> tests/ui/map_flatten_fixable.rs:18:47
    |
 LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_ref).flatten().collect();
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_ref)`
 
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:21:47
+  --> tests/ui/map_flatten_fixable.rs:20:47
    |
 LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(option_id_closure).flatten().collect();
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(option_id_closure)`
 
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:23:47
+  --> tests/ui/map_flatten_fixable.rs:22:47
    |
 LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| x.checked_add(1)).flatten().collect();
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `filter_map` and remove the `.flatten()`: `filter_map(|x| x.checked_add(1))`
 
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:27:47
+  --> tests/ui/map_flatten_fixable.rs:26:47
    |
 LL |     let _: Vec<_> = vec![5_i8; 6].into_iter().map(|x| 0..x).flatten().collect();
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `flat_map` and remove the `.flatten()`: `flat_map(|x| 0..x)`
 
 error: called `map(..).flatten()` on `Option`
-  --> tests/ui/map_flatten_fixable.rs:31:40
+  --> tests/ui/map_flatten_fixable.rs:30:40
    |
 LL |     let _: Option<_> = (Some(Some(1))).map(|x| x).flatten();
    |                                        ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`
 
 error: called `map(..).flatten()` on `Result`
-  --> tests/ui/map_flatten_fixable.rs:35:42
+  --> tests/ui/map_flatten_fixable.rs:34:42
    |
 LL |     let _: Result<_, &str> = (Ok(Ok(1))).map(|x| x).flatten();
    |                                          ^^^^^^^^^^^^^^^^^^^^ help: try replacing `map` with `and_then` and remove the `.flatten()`: `and_then(|x| x)`
 
 error: called `map(..).flatten()` on `Iterator`
-  --> tests/ui/map_flatten_fixable.rs:45:10
+  --> tests/ui/map_flatten_fixable.rs:44:10
    |
 LL |           .map(|n| match n {
    |  __________^
@@ -74,7 +74,7 @@ LL ~         });
    |
 
 error: called `map(..).flatten()` on `Option`
-  --> tests/ui/map_flatten_fixable.rs:66:10
+  --> tests/ui/map_flatten_fixable.rs:65:10
    |
 LL |           .map(|_| {
    |  __________^

